velox/dwio/parquet/writer/Writer.h

```diff
@@ -30,15 +31,66 @@ class ArrowDataBufferSink;
 
 struct ArrowContext;
 
+class DefaultFlushPolicy : public dwio::common::FlushPolicy {
+ public:
+  DefaultFlushPolicy()
+      : rowsInRowGroup_(1'024 * 1'024), bytesInRowGroup_(128 * 1'024 * 1'024) {}
+  DefaultFlushPolicy(uint64_t rowsInRowGroup, int64_t bytesInRowGroup)
+      : rowsInRowGroup_(rowsInRowGroup), bytesInRowGroup_(bytesInRowGroup) {}
+
+  bool shouldFlush(
+      const dwio::common::StripeProgress& stripeProgress) override {
+    return stripeProgress.stripeRowCount >= rowsInRowGroup_ ||
+        stripeProgress.stripeSizeEstimate >= bytesInRowGroup_;
+  }
+
+  void onClose() override {
+    // No-op
+  }
+
+  uint64_t rowsInRowGroup() const {
+    return rowsInRowGroup_;
+  }
+
+  int64_t bytesInRowGroup() const {
+    return bytesInRowGroup_;
+  }
+
+ private:
+  const uint64_t rowsInRowGroup_;
+  const int64_t bytesInRowGroup_;
+};
+
+class LambdaFlushPolicy : public DefaultFlushPolicy {
+ public:
+  explicit LambdaFlushPolicy(
+      uint64_t rowsInRowGroup,
+      int64_t bytesInRowGroup,
+      std::function<bool()> lambda)
+      : DefaultFlushPolicy(rowsInRowGroup, bytesInRowGroup) {
+    lambda_ = lambda;
```

Comment:
Avoid unnecessary copy:
```suggestion
    lambda_ = std::move(lambda);
```